### PR TITLE
feat: add rocksdb Disk cache

### PR DESCRIPTION
**Pull request by lichuang**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

1. introduce new cache type `rockdb` and `rockdbdisk`, while `rockdb` save cache value into rocksdb, and `rockdbdisk` save only cache meta into rocksdb but value into disk.
2. upgrade disk cache version from `v1` to `v2`.

Closes #issue

### CODE REVIEW

File: src/query/storages/common/cache/src/providers/rocksdb_cache.rs [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+// map key -> value
+static KEY2VALUE_COLUMN_PREFIX: &str = "_key2value";
+
+pub struct RocksDbCache {
+    db: rocksdb::TransactionDB,
+    limit: i64,
+}
+
+impl RocksDbCache {
+    pub fn new(path: &str, limit: i64) -> Result<RocksDbCache> {
```
Comment (BohuTANG) [inline src/query/storages/common/cache/src/providers/rocksdb_cache.rs:35]: If we remove the whole path, the rocksdb is work next time?

Comment (lichuang) [reply src/query/storages/common/cache/src/providers/rocksdb_cache.rs:35]: yes, rocksdb will recreate the whole data dir.

**Comment by BohuTANG** [comment]
@dantengsky @lichuang 

BTW, The benchmark CI above is not enable the cache, FYI.

**Comment by dantengsky** [comment]
> @dantengsky @lichuang
> 
> BTW, The benchmark CI above is not enable the cache, FYI.

got it

**Pull request closed without merging**